llvm/lib/Target/AArch64/AArch64Features.td

```diff
@@ -216,8 +211,10 @@ def FeaturePAN_RWV : SubtargetFeature<
 def FeaturePsUAO : SubtargetFeature< "uaops", "HasPsUAO", "true",
     "Enable v8.2 UAO PState (FEAT_UAO)">;
 
-def FeatureCCPP : SubtargetFeature<"ccpp", "HasCCPP",
-    "true", "Enable v8.2 data Cache Clean to Point of Persistence (FEAT_DPB)" >;
+let ArchExtKindSpelling = "AEK_DPB", MArchName = "dpb" in
+def FeatureCCPP : Extension<"ccpp", "CCPP",
+    "Enable v8.2 data Cache Clean to Point of Persistence (FEAT_DPB)", [],
+    "FEAT_DPB", "+ccpp", 190>;
```

Comment:
Making each of these into Extensions will cause `-march=` and `__attribute__((target(...)))` to accept them. That might be a good idea, but I think it requires coordination with GCC. And if that's the intention there should be tests for each case.

Reply:
Ok, I'll explain what the background story is. I am planning to remove the `DependentFeatures` from `ExtensionInfo` in TargetParser because it is duplication of information. We already have the field `Implies` for subtarget features and the TargetParser itself has `ExtensionDependency` to describe 1-1 deps, which can then be expanded with ExtensionSet::enable() to a Bitset. The 1-1 deps can also be tablegen-ed like you already did with the rest of `ExtensionInfo`.

Now, in order use this machiney I described above, all the extensions (even those that don't make their way to the command line, or the attribute target) need to have and extension kind other than AEK_NONE. The only way I see working around it is to add a new field in `ExtensionInfo` to indicate whether an entry is command line facing. This is a bad design if the TargetParser is meant for command line facing extensions. Can you think of a better way?

Reply:
I agree with the goal: we probably want to generate the FMV dependencies (`DependentFeatures`) from the `Implies` field eventually.

`FMVOnlyExtension` already indicates the lack of a `-march` flag is indicated by setting `AEK_NONE`. If you need an `AEK_` to describe the dependencies between extensions, then all `FMVOnlyExtension`s will ultimately need `AEK_` names, and `AEK_NONE` can  no longer be used to indicate that they are "FMV only". So I think you're right that a new field in `ExtensionInfo` is required.